spec/client_spec.rb

```diff
@@ -76,6 +76,55 @@ def redis.with
     context 'when lock is available' do
       after(:each) { lock_manager.unlock(@lock_info) if @lock_info }
 
+      context 'when redis connection error occurs' do
+        let(:servers_with_quorum) {
+          [
+            "redis://#{redis1_host}:#{redis1_port}",
+            "redis://#{redis2_host}:#{redis2_port}",
+            unreachable_redis
+          ]
+        }
+
+        let(:servers_without_quorum) {
+          [
+            "redis://#{redis1_host}:#{redis1_port}",
+            "redis://127.0.0.1:3124",
```

Comment:
how about re-using meaningfully named `unreachable_redis` here too?
```suggestion
            unreachable_redis,
```